consensus/hotstuff/timeoutaggregator/timeout_collectors.go

```diff
@@ -0,0 +1,134 @@
+package timeoutaggregator
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/module/mempool"
+)
+
+// NewCollectorFactoryMethod is a factory method to generate a TimeoutCollector for concrete view
+type NewCollectorFactoryMethod = func(view uint64) (hotstuff.TimeoutCollector, error)
```

Comment:
I think we should allow the factory method to return a `model.ErrViewForUnknownEpoch` as an expected error.

Reply:
answered [here](https://github.com/onflow/flow-go/pull/2740#discussion_r915019387)